R/recover.weaker.R

```diff
@@ -2,172 +2,644 @@
 NULL
 #> NULL
 
+
+#' Custom way of removing duplicate rows from a specifically formatted table.
+#'
+#' @description
+#' Rows are considered as duplicate if the 1st, 2nd and 5th column are less than 1e-10 (tolerance) apart.
+#' Only a single row in this `range` is kept from a group.
+#' @param new.table The table from which the duplicate rows should be removed. Needs at least 5 columns.
+#' Columns 1, 2 and 5 have to be of numeric type.
+#' @param tolerance Tolerance to use for numeric comparisons.
+#' @return Returns the same table with duplicate rows removed.
 #' @export
-duplicate.row.remove <- function(new.table) {
-  new.table <- new.table[order(new.table[, 1], new.table[, 2], new.table[, 5]), ]
+duplicate.row.remove <- function(features, tolerance = 1e-10) {
+  new.table <- features |> dplyr::arrange_at(c("mz", "pos", "area"))
   n <- 1
   m <- 2
   to.remove <- rep(0, nrow(new.table))
 
   while (m <= nrow(new.table)) {
-    if (abs(new.table[m, 1] - new.table[n, 1]) < 1e-10 &
-      abs(new.table[m, 2] - new.table[n, 2]) < 1e-10 &
-      abs(new.table[m, 5] - new.table[n, 5]) < 1e-10) {
+    if (abs(new.table$mz[m] - new.table$mz[n]) < tolerance &
+      abs(new.table$pos[m] - new.table$pos[n]) < tolerance &
+      abs(new.table$area[m] - new.table$area[n]) < tolerance) {
       to.remove[m] <- 1
       m <- m + 1
     } else {
       n <- m
       m <- m + 1
     }
-    # cat("*(", n, m, ")")
   }
 
-  if (sum(to.remove) > 0) new.table <- new.table[-which(to.remove == 1), ]
+  if (sum(to.remove) > 0) {
+    new.table <- new.table[-which(to.remove == 1), ]
```

Comment:
Instead of assigning `1`s to those indices that have to be removed, `to.remove` can just store these indices. This line would then look simply like this:
```R
new.table <- new.table[-to.remove, ]
```